src/db.rs

```diff
@@ -2,154 +2,218 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-use rusqlite::{ self, Connection };
-use std::time::{ SystemTime, UNIX_EPOCH };
-
-#[cfg(test)]
-fn get_db_environment() -> String {
-    "./boxes_test.sqlite".to_string()
-}
-
-#[cfg(not(test))]
-fn get_db_environment() -> String {
-    "./boxes.sqlite".to_string()
-}
+use redis::{ Client, cmd, Connection, ConnectionAddr, ConnectionInfo,
+             pipe, RedisResult };
+use std::time::Duration;
+use std::thread::sleep;
 
 #[derive(RustcEncodable, Debug)]
 pub struct Record {
     pub public_ip: String,
     pub client:    String,
     pub message:   String,
-    pub timestamp: i64 // i64 because of the database type.
-}
-
-fn escape(string: &str) -> String {
-    // http://www.sqlite.org/faq.html#q14
-    string.replace("'", "''")
-}
-
-pub enum FindFilter {
-    PublicIp(String),
-    PublicIpAndClient(String, String)
 }
 
 pub struct Db {
-    // rusqlite::Connection already implements the Drop trait for the
-    // inner connection so we don't need to manually close it. It will
-    // be closed when the UsersDb instances go out of scope.
     connection: Connection
 }
 
 impl Db {
-    pub fn new() -> Db {
-        // TODO: manage errors.
-        let connection = Connection::open(get_db_environment()).unwrap();
-        connection.execute("CREATE TABLE IF NOT EXISTS boxes (
-                public_ip TEXT NOT NULL,
-                client TEXT NOT NULL,
-                message TEXT,
-                timestamp INTEGER
-            )", &[]).unwrap();
-
-        Db {
-            connection: connection
+    pub fn new(db_host: String,
+               db_port: u16,
+               db_password: Option<String>) -> Db {
+        let client = Client::open(ConnectionInfo {
+            addr: Box::new(ConnectionAddr::Tcp(db_host.clone(), db_port)),
+            db: 0,
+            passwd: db_password.clone()
+        }).unwrap();
+
+        loop {
+            match client.get_connection() {
+                Err(err) => {
+                    if err.is_connection_refusal() {
+                        warn!("Could not connect: {} (Will retry)", err);
+                        sleep(Duration::from_millis(1));
```

Comment:
Yes, that's what I intended. Consider it a `setTimeout(..,0);`. This shouldn't be a problem when the Redis server is already running, like in normal usage, as we will connect right away, but because we start the server on the tests, the tests execution race with the server boot up, so I added this to avoid intermittents.